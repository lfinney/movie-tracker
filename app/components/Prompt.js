import React from 'react';
import PropTypes from 'prop-types';

const Prompt = ({ removePopup, userLoginError, dupFav, removeDupPopup }) => {
  return (
    <div className="error-prompt">
      <h2>Whoops!</h2>
      { userLoginError &&
      <h3>Your username and password combination do not match our records.</h3>
      }
      { dupFav &&
        <h3>You already have that moive in your favorites.</h3>
      }
      <h3>Please try again.</h3>
      <button
        className="error-button"
        onClick={ (event) => {
          event.preventDefault();
          removePopup();
          removeDupPopup();
        }}>
        Try Again
      </button>

    </div>
  );
};

Prompt.propTypes = {
  removePopup: PropTypes.func,
  removeDupPopup: PropTypes.func,
  userLoginError: PropTypes.bool,
  dupFav: PropTypes.bool
};

export default Prompt;
