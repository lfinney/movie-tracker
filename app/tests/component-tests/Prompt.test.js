import React from 'react';
import { shallow } from 'enzyme';
import Prompt from '../../components/Prompt';

describe('Prompt', () => {

  it('should render component', () => {
    const wrapper = shallow(<Prompt />);

    expect(wrapper).toMatchSnapshot();
  });
});
