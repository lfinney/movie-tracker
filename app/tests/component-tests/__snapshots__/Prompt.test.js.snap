// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Prompt />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Whoops!
        </h2>,
        <h3>
          Your username and password combination do not match our records.
        </h3>,
        <h3>
          Please try again.
        </h3>,
        <button
          className="error-button"
          onClick={[Function]}
        >
          Try Again
        </button>,
      ],
      "className": "error-prompt",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Whoops!",
        },
        "ref": null,
        "rendered": "Whoops!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Your username and password combination do not match our records.",
        },
        "ref": null,
        "rendered": "Your username and password combination do not match our records.",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Please try again.",
        },
        "ref": null,
        "rendered": "Please try again.",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Try Again",
          "className": "error-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Try Again",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Whoops!
          </h2>,
          <h3>
            Your username and password combination do not match our records.
          </h3>,
          <h3>
            Please try again.
          </h3>,
          <button
            className="error-button"
            onClick={[Function]}
          >
            Try Again
          </button>,
        ],
        "className": "error-prompt",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Whoops!",
          },
          "ref": null,
          "rendered": "Whoops!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Your username and password combination do not match our records.",
          },
          "ref": null,
          "rendered": "Your username and password combination do not match our records.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Please try again.",
          },
          "ref": null,
          "rendered": "Please try again.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Try Again",
            "className": "error-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Try Again",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
